opentracing-impl/src/main/java/io/opentracing/impl/AbstractActiveSpan.java

```diff
@@ -0,0 +1,158 @@
+package io.opentracing.impl;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * {@link AbstractActiveSpan} deals with {@link ActiveSpanSource}/{@link ActiveSpan}/{@link Continuation}
+ * reference counting while allowing subclasses to extend with bindings to ThreadLocals, etc.
+ */
+public abstract class AbstractActiveSpan implements ActiveSpan {
+    private final AtomicInteger refCount;
+    protected final Span wrapped;
+
+    protected AbstractActiveSpan(Span wrapped, AtomicInteger refCount) {
+        this.wrapped = wrapped;
```

Comment:
To do: `null`-checks before setting `final` fields?
also makes it clear whether wrapped is nullable
maybe initialize refCount to new AtomicInteger(1) by default in case of null?
Validate current refCount value in constructor?